### PR TITLE
migrated confirm_modal.jsx to tsx

### PR DESCRIPTION
**Pull request by Saamiyah**
#### Summary
Migrating confirm_modal.jsx to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12453

### CODE REVIEW

File: components/confirm_modal.tsx [hunk 1]
```diff
@@ -1,78 +1,83 @@
...
+
+    /*
+     * Set to hide the cancel button
+     */
+    hideCancel: boolean;
+};
+
+export default class ConfirmModal extends React.PureComponent<Props> {
+
+    public constructor(props: Props) {
```
Comment (reflog) [inline components/confirm_modal.tsx]: empty constructor is redundant

**Comment by reflog** [comment]
please rebase with master and fix the tests (maybe updating snapshots is needed)

**Comment by jasonblais** [comment]
Hi @Saamiyah, big thanks for your contribution! Let us know if you have any questions about rebasing against the `master` branch and resolving the build issues -- we'd be happy to help you!

**Comment by hanzei** [comment]
Hi @Saamiyah,

Given that we have not heard from you for a long time, I'm going to close this PR. Thanks for all the time you put into it.

**Pull request closed without merging**